rcon/battleye/client.py

```diff
@@ -55,10 +56,11 @@ def receive(self, max_length: int = 4096) -> Response:
 
     def communicate(self, request: Request) -> Response:
         """Send a request and receive a response."""
-        with self._socket.makefile('wb') as file:
-            file.write(bytes(request))
+        with lock:
```

Comment:
Should really all instances of `Client` share the same lock?
Would it be better or worse, if each instance of `Client` would have its own lock?

Reply:
I vote for a per-instance lock.
E.g. why should clientA wait for clientB if it is contacting an entirely different server?

Reply:
good point!